### ISSUE TITLE
VoteParty

### ISSUE THREAD

**Issue by ConnorFranta** [BenCodez/VotingPlugin#177]
When using vote party, I have come across two issues with real votes.
I have it set up to run two reward files. One for a broadcast, another with a vote delay and the reward.

1. When the total is reached, sometimes the files won't run, other times it will run up to five times.
2. The delay is set to one minute, but it can take up to five minutes to run. _(I can live with this but I'm guessing this is because of the DelayBetweenUpdate?)_

My troubleshooting steps:
- Run each reward individually (/av GiveReward) Runs fine. Delay is still up to 5 minutes.
- Run fake votes (/av vote) Unable to duplicate the problem. Delay is still up to 5 minutes.
- Live test with real players voting. (This is when it will run more than once or not at all) Delay is still up to 5 minutes.

Thanks in advance!

[Reports.1489826982657.zip](https://github.com/Ben12345rocks/VotingPlugin/files/852264/Reports.1489826982657.zip)

**Comment by BenCodez** [comment]
2. There is a timer in the background that checks the timed/delayed rewards, there is no option to change what the timer is set to (Might change this), its set to 5 minutes right now, could add an option for that in the future.

/av vote runs the same code as an actual vote, not sure how it can differ.

Turn debug on, recreate the problem and send the log so I can see what's happening.

Won't have much time until later in the day to really debug this, as far as I know it should work fine.

**Comment by BenCodez** [comment]
Looks like vote party is working perfectly for a couple of people, still haven't had time to test it myself, but I thought I mention that

**Comment by ConnorFranta** [comment]
I believe I might have figured it out.

1. "GiveAllPlayers:" under VoteParty in the config was 'true' so the reward file would go to all players.
2. When the total was reached, the reward file ran, "RewardType:" wasn't set, so that would give the reward to "BOTH" which is all players, including the offline players. When offline players would rejoin, later on, the file would re-trigger, run the broadcast file, then run the command, which was a give all.

I have now set GiveAllPlayers: false; RewardType: ONLINE

I am running the debug now, waiting for the vote party total to reach.

**Comment by BenCodez** [comment]
Use a message instead of broadcast as well.

On Mar 18, 2017 6:01 PM, "ConnorFranta" <notifications@github.com> wrote:

I believe I might have figured it out.

   1. "GiveAllPlayers:" under VoteParty in the config was 'true' so the
   reward file would go to all players.
   2. When the total was reached, the reward file ran, "RewardType:" wasn't
   set, so that would give the reward to "BOTH" which is all players,
   including the offline players. When offline players would rejoin, later on,
   the file would re-trigger, run the broadcast file, then run the command,
   which was a give all.

I have now set GiveAllPlayers: false; RewardType: ONLINE

I am running the debug now, waiting for all the vote total to reach.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/Ben12345rocks/VotingPlugin/issues/177#issuecomment-287577999>,
or mute the thread

**Comment by ConnorFranta** [comment]
Debug
[Log.txt](https://github.com/Ben12345rocks/VotingPlugin/files/852777/Log.txt)

**Comment by BenCodez** [comment]
Mind sending the server log? Did it work? Can't do much right now or for
awhile, probably till tomorrow until I get on my PC

**Comment by ConnorFranta** [comment]
I don't have the authority to release the server logs, the server owner wouldn't allow it. It also contains our player IP's. Not that I don't trust you, it's just vital information. We vow to our members we will never release their information to anyone, including staff. 

The broadcast ran more than once when the total was reached, but I am guessing that was due to the file running for each person. I changed out the broadcast to run to the player and also changed out the giveall command to give @ %player% and everyone only received one this time. So it should be good to go now.
